src/main/java/com/minecolonies/util/BlockPosUtil.java

```diff
@@ -142,11 +142,11 @@ public static boolean isClose(@NotNull BlockPos coordinate, @NotNull EntityCitiz
      * @param block2 position two.
      * @return squared distance.
      */
-    public static int getDistanceSquared(@NotNull BlockPos block1, @NotNull BlockPos block2)
+    public static long getDistanceSquared(@NotNull BlockPos block1, @NotNull BlockPos block2)
     {
-        int i = block1.getX() - block2.getX();
-        int i1 = block1.getY() - block2.getY();
-        int i2 = block1.getZ() - block2.getZ();
+        long i = block1.getX() - block2.getX();
+        long i1 = block1.getY() - block2.getY();
+        long i2 = block1.getZ() - block2.getZ();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Cast one of the operands of this substraction operation to a "long". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2184)